ws-calculator/src/main/java/org/egov/wsCalculation/builder/WSCalculatorQueryBuilder.java

```diff
@@ -0,0 +1,195 @@
+	private final static String Query = "SELECT mr.id, mr.connectionNo as connectionId, mr.billingPeriod, mr.meterStatus, mr.lastReading, mr.lastReadingDate, mr.currentReading, "
+			+ "mr.currentReadingDate FROM meterreading mr";
+
+	private final static String noOfConnectionSearchQuery = "SELECT count(*) FROM meterreading WHERE";
+    
+	private final static String noOfConnectionSearchQueryForCurrentMeterReading= "select mr.currentReading from meterreading mr";
+	
+	private final static String tenentIdWaterConnectionSearchQuery="select DISTINCT tenantid from connection";
+	
+	private final static String connectionNoWaterConnectionSearchQuery = "SELECT conn.connectionNo as conn_no FROM water_service_connection wc INNER JOIN connection conn ON wc.connection_id = conn.id";
+	
+	private static final String connectionNoListQuery = "SELECT distinct(connectionno) FROM connection ws";
+
+	private static final String distinctTenantIdsCriteria = "SELECT distinct(tenantid) FROM connection ws";
+
+
+	public String getDistinctTenantIds() {
+		StringBuilder query = new StringBuilder(distinctTenantIdsCriteria);
+		log.info("Query : " + query);
+		return query.toString();
+	}
+	/**
+	 * 
+	 * @param criteria
+	 *            would be meter reading criteria
+	 * @param preparedStatement
+	 * @return Query for given criteria
+	 */
+	public String getSearchQueryString(MeterReadingSearchCriteria criteria, List<Object> preparedStatement) {
+		StringBuilder query = new StringBuilder(Query);
+		String resultantQuery = Query;
+		boolean isAnyCriteriaMatch = false;
+		if (!criteria.getConnectionNos().isEmpty()) {
+			addClauseIfRequired(preparedStatement, query);
+			query.append(" mr.connectionNo IN (").append(createQuery(criteria.getConnectionNos())).append(" )");
+			addToPreparedStatement(preparedStatement, criteria.getConnectionNos());
+			isAnyCriteriaMatch = true;
+		}
+		if(isAnyCriteriaMatch == false)
+			return null;
+		resultantQuery = query.toString();
+		resultantQuery = addOrderBy(resultantQuery);
+		if (query.toString().indexOf("WHERE") > -1)
+			resultantQuery = addPaginationWrapper(resultantQuery, preparedStatement, criteria);
+		return resultantQuery;
+	}
+
+	private String createQuery(Set<String> ids) {
+		StringBuilder builder = new StringBuilder();
+		int length = ids.size();
+		for (int i = 0; i < length; i++) {
+			builder.append(" ?");
+			if (i != length - 1)
+				builder.append(",");
+		}
+		return builder.toString();
+	}
+
+	private void addToPreparedStatement(List<Object> preparedStatement, Set<String> ids) {
+		ids.forEach(id -> {
+			preparedStatement.add(id);
+		});
+	}
+
+	private void addClauseIfRequired(List<Object> values, StringBuilder queryString) {
+		if (values.isEmpty())
+			queryString.append(" WHERE ");
+		else {
+			queryString.append(" AND");
+		}
+	}
+
+	private String addPaginationWrapper(String query, List<Object> preparedStmtList,
+			MeterReadingSearchCriteria criteria) {
```

Comment:
This will not work for one to many joins, please revisit this if that's the case